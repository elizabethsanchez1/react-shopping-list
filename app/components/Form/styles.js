import { create } from 'react-native-platform-stylesheet';
import colors from '../../config/colors';
import theme from '../../styles/theme.style';

export default create({


  // global styles
  fontFamily: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
  },

  fontWeight: {
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
  },







  // button specific styles

  primaryButton: {
    marginTop: 20,
    backgroundColor: theme.PRIMARY_COLOR,
    borderRadius: 4,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
  },



  secondaryButton: {
    // marginVertical: 20,
    // marginHorizontal: 30,
    padding: 12,
    borderWidth: 1,
    borderColor: theme.BORDER_COLOR,
    borderRadius: 5,
  },
  secondaryButtonText: {
    alignSelf: 'center',
    color: theme.PRIMARY_FONT_COLOR,
  },




  disabledStyle: {
    backgroundColor: theme.PRIMARY_COLOR,
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0,
    shadowRadius: 0,
    elevation: 0,
  },

  disabledText: {
    color: theme.DISABLED_TEXT_COLOR,
  },

  // Floating button specific styles

  actionButton: {
    width: 60,
    height: 60,
    borderRadius: 50,
    padding: 5,
    paddingBottom: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.5,
    shadowRadius: 2,
    elevation: 1,
  },



  // Input specific styles

  input: {
    width: 'auto',
    height: 40,
    marginLeft: 0,
    marginRight: 0,
    marginBottom: 0,
    borderBottomColor: theme.BORDER_COLOR,
    borderBottomWidth: theme.BORDER_BOTTOM_WIDTH,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },


  viewContainer: {
    marginLeft: 0,
    marginRight: 0,
    borderBottomWidth: 0,
    marginBottom: 10,
  },

  inputContainer: {
    marginLeft: 0,
    marginRight: 0,
    borderBottomWidth: 0,
    marginBottom: 0,
  },


  errorMessage: {
    marginLeft: 0,
    marginRight: 0,
    marginTop: 10,
    color: theme.ERROR_RED,
  },


  errorMessageContainer: {
    marginLeft: 0,
    marginRight: 0,
  },




  basicInput: {
    borderBottomColor: theme.BORDER_COLOR,
    borderBottomWidth: theme.BORDER_BOTTOM_WIDTH,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    height: 40,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    marginBottom: 15,
    color: theme.PRIMARY_FONT_COLOR,
  }
});
