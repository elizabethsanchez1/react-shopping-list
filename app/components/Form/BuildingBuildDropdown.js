import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import { Dropdown } from 'react-native-material-dropdown';
import styles from './styles';
import theme from '../../styles/theme.style';

const BuildingBuildDropdown = (props) => {
  if (props.dropdown1Data.length > 1) {
    return (
      <View style={{flexDirection: 'row', paddingLeft: 20, paddingRight: 20}}>
        <Dropdown
          label="Copy from"
          placeholderTextColor="white"
          baseColor='white'
          textColor='white'
          selectedItemColor='black'
          containerStyle={{width: '50%', paddingRight: 15}}
          data={props.dropdown1Data}
          onChangeText={value => props.onChange({copyFrom: value})}
        />
        <Dropdown
          label="Copy to"
          placeholderTextColor="white"
          baseColor='white'
          textColor='white'
          selectedItemColor='black'
          containerStyle={{width: '50%', paddingLeft: 15}}
          data={props.dropdown2Data}
          onChangeText={value => props.onChange({copyTo: value})}
        />
      </View>
    );
  }

  return null;
};

BuildingBuildDropdown.propTypes = {
  dropdown1Data: PropTypes.array,
  dropdown2Data: PropTypes.array,
  onChange: PropTypes.func,
};

export default BuildingBuildDropdown;
