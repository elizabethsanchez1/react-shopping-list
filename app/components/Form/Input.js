import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import { FormLabel, FormInput, FormValidationMessage } from 'react-native-elements';
import styles from './styles';
import theme from '../../styles/theme.style';

const Input = React.forwardRef((props, ref) => {
  return (
    <View style={[styles.viewContainer ,props.containerStyling]}>
      {
        (props.label !== undefined) &&
          <FormLabel>{props.label}</FormLabel>
      }
      <FormInput
        ref={ref}
        autoCapitalize="none"
        autoCorrect={false}
        containerStyle={styles.inputContainer}
        inputStyle={[styles.input, props.inputStyling]}
        placeholderTextColor='#A1A1A1'
        {...props}
      />
      {
        (props.errorMessage !== undefined) &&
          <FormValidationMessage labelStyle={styles.errorMessage}>
            {props.errorMessage}
          </FormValidationMessage>
      }
    </View>
  );
});


Input.propTypes = {
  label: PropTypes.any,
  containerStyling: PropTypes.object,
  errorMessage: PropTypes.string,
};

export default Input;
