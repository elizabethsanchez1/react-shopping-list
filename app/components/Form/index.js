import Input from './Input';
import BuildingBuildDropdown from './BuildingBuildDropdown';

export { Input, BuildingBuildDropdown };
