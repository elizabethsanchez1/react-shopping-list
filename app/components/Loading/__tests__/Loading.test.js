import 'react-native';
import React from 'react';
import { Input } from '../../Form';
import renderer from 'react-test-renderer';
import { Loading } from "../index";


describe( 'Loading Component', () => {


  it( 'Loading base visually looks the same', () => {
    const tree = renderer.create(
      <Loading />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

} );




