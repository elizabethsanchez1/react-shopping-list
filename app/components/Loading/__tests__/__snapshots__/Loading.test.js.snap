// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Loading base visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#47474D",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#72DEFF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
